### PR TITLE
fix(remix-react): fix `<Form>` submit to not break `formMethod` functionality

### PR DESCRIPTION
**Pull request by kevlened**
Closes: #2162 and #3613 (supersedes both) – credits to @mdoury and @nrako

> the `<Form>` component – unlike native `<form>` - does not respect the `formmethod` attribute set
> on the submitter (the button submitting the form).

* [x]  ~Docs~ N/A
* [x]  Tests

Testing Strategy:

```js
<Form method="post">
  <input name="eat" value="pizza" />
  <button type="submit"> Submit with GET </button>
</Form>
// should submit with a GET to `/?eat=pizza` but instead do a POST
```

See integration [Bug-Report-Test](https://github.com/remix-run/remix/blob/37b0140f80dfb333240da96fb9ff83e713848458/integration/bug-report-test.ts) via commit: [37b0140](https://github.com/remix-run/remix/commit/37b0140f80dfb333240da96fb9ff83e713848458)

**test: failing test with `<Form>` not respecting `formMethod`** [06333c1](https://github.com/remix-run/remix/commit/06333c1ed3e3fe1a1950013d663a9f1041dbc8a5) <Nicholas Rakoto> Thu, 30 Jun 2022 00:33:15 +0200

```
Bug report integration test demonstrating how the `<Form>` component – 
unlike native `<form>` - does not respect the `formmethod` attribute set on
the submitter (the `<button>` submitting the form).
```

**fix(remix-react): submit `<Form>` w/method set by submitter `formmethod`** [bfd1f88](https://github.com/remix-run/remix/commit/bfd1f88de007ba79d1dca85391f1a3b7c58d3677) <Nicholas Rakoto> Thu, 30 Jun 2022 00:32:22 +0200

```
Fix `<Form>` component by not passing the form's `method` into the
`options` when calling the `submit(target, options)` function (created with
the `useSubmit()` hook). This ensures that the option does not take 
precedence on any `formethod` set on the submitter which otherwise break the
`formmethod` functionality.

Since #3094 and #3094 the `<Form>` passes the "submitter" (if any) as the
`target` to the `submit(target, options)` (`useSubmitImpl`) which will
attempt to read the `formmethod`, `formaction` and `formenctype` attributes
on the target. Therefore, the responsability to infer which method should be
used should solely be on `useSubmitImpl` for the given
`target`.

Co-authored-by: Maxime Doury <maxime.doury@docavenue.com>
```

### CODE REVIEW

File: integration/form-test.ts
Comment (MichaelDeBoey) [inline integration/form-test.ts:306]:
```suggestion
          import { json } from '@remix-run/node'
```

File: integration/form-test.ts
Comment (MichaelDeBoey) [inline integration/form-test.ts:867]: Let's either create 1 test that checks both or wrap this with a `'overrides the form "method" attribute with the button "formmethod" attribute'` `describe`.

File: integration/form-test.ts
Comment (MichaelDeBoey) [inline integration/form-test.ts:335]:
```suggestion
          import { Form, useActionData } from "@remix-run/react";
          import { json } from "@remix-run/node";

```

File: integration/form-test.ts
Comment (MichaelDeBoey) [inline integration/form-test.ts:353]:
```suggestion
          import { Form, useActionData } from "@remix-run/react";
          import { json } from "@remix-run/node";

```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: integration/form-test.ts
Comment (kevlened) [reply integration/form-test.ts]: I've wrapped them in a `describe`, as suggested

File: integration/form-test.ts
Comment (MichaelDeBoey) [inline integration/form-test.ts:884]:
```suggestion
    test('uses the button `formmethod` attribute', async ({ page }) => {
```

File: integration/form-test.ts
Comment (kevlened) [reply integration/form-test.ts]: fixed to also include add double, rather than single quotes around the full text

File: integration/form-test.ts
Comment (kevlened) [reply integration/form-test.ts]: fixed to also include add double, rather than single quotes around the full text

**Review by MichaelDeBoey — APPROVED** [review]

**Review by mcansh — APPROVED** [review]

**Comment by kevlened** [comment]
That’s odd. I’m confident the functionality is broken, but I was able to work around it. I’ll create a repro example, then try to see why the tests aren’t catching it.

**Comment by evanstern** [comment]
I can also confirm that this issue still occurs for me in the current release.

**Comment by mian-muhammad** [comment]
I am facing the same issue in the current release.

**Comment by kiliman** [comment]
Until PR #4053 is merged in, you can use `patch-package`. It's literally a single line change. Copy this patch to your _patches_ folder.

https://gist.github.com/kiliman/5a8dac031da8d545dc293fdcf82e9135

**Comment by mcansh** [comment]
> Until PR #4053 is merged in, you can use `patch-package`. It's literally a single line change. Copy this patch to your _patches_ folder.
> 
> https://gist.github.com/kiliman/5a8dac031da8d545dc293fdcf82e9135

there's two solutions to this problem, the one provided by that PR and [this](https://github.com/remix-run/remix/commit/58f1618e6d8ef6ca17771294087c19dc8b8ce8f5#diff-594d4b6ed5bd4abff7e9319566b5505b6a85753e665ff4d2113e888384a98377). the issue is due to defaulting `Form` to `method=get`, which we should, however we check for options.method (and friends) lower in the component where options.method is always set as it has a default value in `Form`. I still lean toward the fix in #4053, but I'll leave it up to discussion

**Review by mcansh — APPROVED** [review]

**Pull request merged**
